blog/2024-04-30-opentofu-1-7-0.md

```diff
@@ -0,0 +1,46 @@
+slug: opentofu-1-7-0
+description: OpenTofu 1.7.0 is now available with full state encryption, dynamic provider-defined functions, the removed and loopable import blocks, new migration guides, and much more.
+image: /img/blog/opentofu-1-7-0.png
+---
+
+In the last few months [since our first stable release](/blog/opentofu-is-going-ga/) the OpenTofu community and the core team have worked hand in hand to bring functionality to OpenTofu that has been requested for years.  We are proud to announce the immediate availability of [OpenTofu 1.7.0](https://github.com/opentofu/opentofu/releases/tag/v1.7.0), bringing you the following highlights:
+
+- [**Full state encryption**](/docs/intro/whats-new/#state-encryption) protects your state file from prying eyes, no matter what storage backend you use. You can provide your encryption keys securely using environment variables, or using a key management system such as AWS KMS, GCP KMS, or OpenBao. *This feature has been developed in collaboration with Stephan Bartels (Interhyp) and Alex Scheel from the OpenTofu community, whom we would like to thank for their work on this feature.*
+- [**Dynamic provider-defined functions**](/docs/intro/whats-new/#provider-defined-functions) let a provider not only provide resources, but also native functions you can use in your OpenTofu code. What's more, we added an OpenTofu-only feature to let providers dynamically define custom functions based on your configuration. This enhancement allows you to fully integrate other programming languages as [shown in our live stream](https://www.youtube.com/watch?v=6OXBv0MYalY). You can try out this functionality with our experimental [Lua](https://github.com/opentofu/terraform-provider-lua) and [Go](https://github.com/opentofu/terraform-provider-go) providers.
+- [**The removed block**](/docs/intro/whats-new/#removed-block) lets you mark OpenTofu-created resources for removal from the state file, but still keep the infrastructure you created.
+- [**Loopable import blocks**](/docs/intro/whats-new/#loopable-import-blocks) let you bulk-import resources declaratively in your OpenTofu code, helping with large-scale migrations.
+
+As with the previous version, OpenTofu remains a drop-in replacement for its predecessor Terraform 1.5.5 and has easy migration paths from later versions. Check out the overhauled [**migration guides**](/docs/intro/migration/) for detailed migration instructions. You can find the full list of changes and comprehensive examples [in the documentation](/docs/intro/whats-new/).
+
+## The OpenTofu community grew significantly
+
+Since the first release, the OpenTofu community and adoption have been growing significantly. While we don't believe in tracking our users and don't have accurate numbers, we have seen a constant month-over-month growth. Over just the last months our registry usage has doubled to well over million requests per day.
```

Comment:
```suggestion
Since the first release, the OpenTofu community and adoption have been growing significantly. While we don't believe in tracking our users and don't have accurate numbers, we have seen a constant month-over-month growth. Over just the last months our registry usage has doubled to well over a million requests per day.
```